### DISCUSSION TITLE
Update the webapp after each successful project index

### DISCUSSION THREAD

**Discussion by tchikh411** [oracle/opengrok#4682] [General]
**Description**
We run the index everynight for 700 projects with total size of 5TB. This takes hours to index and the configuration is written to webapp only after the full index completes. It would be good if the webapp configuration is updated after every successful index. 

We are running the index with:
`${JAVA_HOME}/bin/java" \
   -Djava.util.logging.config.file="${OPENGROK_LOGGER_CONFIG_PATH}" \
   ${JAVA_OPTS} \
   -jar "${OPENGROK_INSTANCE_BASE}/lib/opengrok.jar" \
   --projects \
   --verbose \
   ${HISTORY_ENABLED:+-H} \
   --source "${OPENGROK_INSTANCE_BASE}/src" \
   --dataRoot "${OPENGROK_INSTANCE_BASE}/data" \
   ${RENAMED_HISTORY:+--renamedHistory} ${RENAMED_HISTORY} \
   ${FOLLOW_SYMLINKS} \
   ${OPENGROK_IGNORE_PATTERNS} \
   ${OPENGROK_INCLUDE_PATTERNS} \
   ${EXUBERANT_CTAGS+--ctags "${EXUBERANT_CTAGS}"} \
   ${OPENGROK_CTAGS_OPTIONS_FILE:+--ctagOpts             "${OPENGROK_CTAGS_OPTIONS_FILE}"} \
   --search \
   --assignTags \
   ${OPENGROK_READONLY_XML_CONFIGURATION:+-R            "${OPENGROK_READONLY_XML_CONFIGURATION}"} \
   --writeConfig "${OPENGROK_INSTANCE_BASE}/etc/configuration.xml" \
   ${OPENGROK_USER_LOOKUP:+--userPage} ${OPENGROK_USER_LOOKUP} \
   --uri "${WEBAPP_CONFIG_URI}"`

**Comment by vladak** [comment]
You are looking for [per project workflow](https://github.com/oracle/opengrok/wiki/Per-project-management-and-workflow).
